server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepository.java

```diff
@@ -142,4 +142,10 @@
      */
     boolean isAdministrator(String username) throws UsersRepositoryException;
 
+    /**
+     * @return true if one can use {@link UsersRepository#updateUser(User)} {@link UsersRepository#addUser(String, String)}
```

Comment:
it's not complexity, it's work.
Making things modular rather than runtime checks lower complexity.
But you're right about it's a price to pay : do we want to pay ? opinions ?